### PR TITLE
Feature/add tag component

### PR DESCRIPTION
**Pull request by enguerranws**

### CODE REVIEW

**Comment by enguerranws** [comment]
Just pushed a renaming and reviewed quickly. 

Stories are running fine. 

@garronej I just had a doubt on how we should handle _onDismiss_ callback, and if we should provide one. I think onClick is more generic and cover the onDismiss need, but let me know.

**Comment by garronej** [comment]
Thanks for this,  
If you have a doubt, let's schedule a peer progaming session.

File: stories/Tag.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        "isSmall": {
+            "control": { "type": "boolean" }
+        },
+        "iconId": {
+            "options": (() => {
+                const options = ["fr-icon-checkbox-circle-line", "ri-ancient-gate-fill"] as const;
+
+                assert<
+                    typeof options[number] extends NonNullable<TagProps["iconId"]> ? true : false
+                >();
```
Comment (garronej) [inline stories/Tag.stories.tsx:25]: I would like to see an assert<Equal<>>() here, ont an extends.

**Comment by enguerranws** [comment]
Ready for review

NB:
- heavily inspired by the Button component
- depending on the props, it can be a `button`, `a` or a `span`
- added type test

**Comment by garronej** [comment]
> Ready for review
> 
> NB:
> 
> * heavily inspired by the Button component
> * depending on the props, it can be a `button`, `a` or a `span`
> * added type test

I'm loving it!

File: src/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                iconId && "fr-tag--icon-left", // actually, it's always left but we need it in order to have the icon rendering
+                isDismissible && "fr-tag--dismiss"
+            ),
+            linkProps !== undefined && linkProps.className,
+            prop_className
+        );
+
+        return (
+            <>
+                {linkProps && (
```
Comment (garronej) [inline src/Tag.tsx:113]:
```suggestion
                {linkProps !== undefined && (
```
We never use non boolean value as if they where boolean, it makes the code harder to reason about and it's the source of countless bugs.

File: src/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                            ...linkProps?.style,
+                            ...style
+                        }}
+                        ref={ref as React.ForwardedRef<HTMLAnchorElement>}
+                        {...rest}
+                    >
+                        {children}
+                    </Link>
+                )}
+                {nativeButtonProps && (
```
Comment (garronej) [inline src/Tag.tsx:128]:
```suggestion
                {nativeButtonProps !== undefined && (
```
Again

File: src/Tag.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                        onClick={onClick ?? nativeButtonProps?.onClick}
+                        disabled={nativeButtonProps?.disabled}
+                        ref={ref as React.ForwardedRef<HTMLButtonElement>}
+                        aria-pressed={pressed}
+                        {...rest}
+                    >
+                        {children}
+                    </button>
+                )}
+                {!linkProps && !nativeButtonProps && (
```
Comment (garronej) [inline src/Tag.tsx:146]: Same

File: test/types/Tag.tsx
Comment (garronej) [inline test/types/Tag.tsx:1]: Thank you for this!

**Review by garronej — COMMENTED** [review]
Thank you for your work

File: test/types/Tag.tsx
Comment (enguerranws) [reply test/types/Tag.tsx:1]: Fixed all of this, tested it with type test and checked on storybook

**Comment by garronej** [comment]
Nice! Thank you very much!

**Pull request merged**